### PR TITLE
Stats default fields

### LINKED ISSUE #5450: Include hostname in all `_node` API outputs
Currently the output is not consistent, e.g. the `_node/hot_threads` includes the hostname in the output, while `_node/stats` doesn't.

Example:

```
curl http://localhost:9601/_node/hot_threads?pretty
{
  "hostname" : "LS5-Test",
...
```
```
curl http://localhost:9601/_node/stats?pretty
{
  "events" : {
...
```

It could be useful to have the have the hostname in the output, e.g. for monitoring that calls several instances etc.

### PR DESCRIPTION
**Pull request by andrewvc**
This fixes #5450 and better conforms the API to #5475 

All API endpoints now return the following three fields:

* host
* http_address
* version

### CODE REVIEW

File: logstash-core/lib/logstash/api/commands/default_metadata.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require "logstash/api/commands/base"
+
+module LogStash
+  module Api
+    module Commands
+      class DefaultMetadata < Commands::Base
```
Comment (untergeek) [inline logstash-core/lib/logstash/api/commands/default_metadata.rb]: I like this change!

File: logstash-core/spec/api/lib/api/support/resource_dsl_methods.rb [hunk 2]
```diff
@@ -23,6 +23,21 @@ def test_api(expected, path)
...
+        it "should include the host" do
+          expect(payload["host"]).to eql(Socket.gethostname)
+        end
+
+        it "should include the version" do
+          expect(payload["version"]).to eql(LOGSTASH_CORE_VERSION)
+        end
+
+        it "should include the http address" do
+          expect(payload["http_address"]).to eql("#{Socket.gethostname}:9600")
```
Comment (suyograo) [inline logstash-core/spec/api/lib/api/support/resource_dsl_methods.rb]: is there a way to not hardcode `9600` here in the `expect`. wondering if we can use the default constant which is defined in the core

File: logstash-core/lib/logstash/api/commands/system/basicinfo_command.rb [hunk 3]
```diff
@@ -9,12 +9,7 @@ module System
         class BasicInfo < Commands::Base
 
           def run
-            {
-              "hostname" => hostname,
-              "version" => {
-                "number" => LOGSTASH_VERSION
-              }
-            }
+            {} # Just return the default metadata
```
Comment (untergeek) [inline logstash-core/lib/logstash/api/commands/system/basicinfo_command.rb]: This cleans the code up nicely.

**Comment by untergeek** [comment]
LGTM™  I like these changes!

File: logstash-core/spec/api/lib/api/support/resource_dsl_methods.rb (hunk 2 above)
Comment (untergeek) [inline logstash-core/spec/api/lib/api/support/resource_dsl_methods.rb]: Oooh, hadn't thought of that.  Yeah, we should not hard-code the 9600 if at all possible.

File: logstash-core/lib/logstash/api/commands/base.rb [hunk 4]
```diff
@@ -8,22 +8,18 @@ def initialize(service = LogStash::Api::Service.instance)
...
 
-        def hostname
-          service.agent.node_name
-        end
-
         def uptime
           service.agent.uptime
         end
-
+
```
Comment (jordansissel) [inline logstash-core/lib/logstash/api/commands/base.rb]: unintentional whitespace change?

File: logstash-core/lib/logstash/api/commands/base.rb [hunk 5]
```diff
@@ -8,22 +8,18 @@ def initialize(service = LogStash::Api::Service.instance)
...
+        
         def started_at
           (LogStash::Agent::STARTED_AT.to_f * 1000.0).to_i
         end
 
         def extract_metrics(path, *keys)
           service.extract_metrics(path, *keys)
         end
-      end
+      end
```
Comment (jordansissel) [inline logstash-core/lib/logstash/api/commands/base.rb]: unintentional whitespace change?

File: logstash-core/lib/logstash/api/commands/node.rb [hunk 6]
```diff
@@ -8,8 +8,8 @@ def all
           {
             :pipeline => pipeline,
             :os => os,
-            :jvm => jvm
-          }           
+            :jvm => jvm
```
Comment (jordansissel) [inline logstash-core/lib/logstash/api/commands/node.rb]: whitespace?

File: logstash-core/lib/logstash/api/commands/node.rb [hunk 7]
```diff
@@ -8,8 +8,8 @@ def all
           {
             :pipeline => pipeline,
             :os => os,
-            :jvm => jvm
-          }           
+            :jvm => jvm         
+          }
```
Comment (jordansissel) [inline logstash-core/lib/logstash/api/commands/node.rb]: whitespace?

File: logstash-core/spec/api/spec_helper.rb [hunk 8]
```diff
@@ -92,7 +94,7 @@ def do_request(&block)
     @runner = LogStashRunner.new
     @runner.start
   end
-
+
```
Comment (jordansissel) [inline logstash-core/spec/api/spec_helper.rb]: whitespaceeeeeeeee added?

File: logstash-core/lib/logstash/api/commands/base.rb (hunk 4 above)
Comment (andrewvc) [inline logstash-core/lib/logstash/api/commands/base.rb]: My editor put it in but maybe it makes sense? White space should be indented too!

File: logstash-core/lib/logstash/api/commands/base.rb (hunk 5 above)
Comment (andrewvc) [inline logstash-core/lib/logstash/api/commands/base.rb]: Yep this one should go

**Comment by jordansissel** [comment]
```
% curl 'http://192.168.1.124:9600/_node?pretty'
% curl -s 'http://192.168.1.124:9600/_node?pretty' 
{
  "host" : "localhost.localdomain",
  "version" : "5.0.0.dev",
  "http_address" : "http://0.0.0.0:9600",
...
```

LGTM

File: logstash-core/lib/logstash/api/commands/node.rb (hunk 7 above)
Comment (andrewvc) [inline logstash-core/lib/logstash/api/commands/node.rb]: This was good, removing the extra spaces

File: logstash-core/spec/api/spec_helper.rb (hunk 8 above)
Comment (andrewvc) [inline logstash-core/spec/api/spec_helper.rb]: This brings the indentation inline. Isn't that good?

**Comment by suyograo** [comment]
LGTM

File: logstash-core/spec/api/spec_helper.rb (hunk 8 above)
Comment (jordansissel) [inline logstash-core/spec/api/spec_helper.rb]: :panda_face:

**Pull request closed without merging**